packages/social-metadata-previews/src/helpers/truncation.js

```diff
@@ -0,0 +1,41 @@
+// Which symbol should be used for indicating that data was truncated.
+const TRUNCATION_SYMBOL = "...";
+
+// How many characters fit on one line.
+// This is an approximation because actually calculating the width of an (rendered) element is a lot of work.
+const MAX_CHARS = 148;
+const MAX_CHARS_ONE_LINE = MAX_CHARS / 2;
+
+/**
+ * Truncates a piece of text to a predefined. It also adds a truncation symbol to indicate that truncation happened.
+ *
+ * The length of truncation is based on two variables:
+ *  - type If the type is title, we always truncate the text to one line.
+ *  - large If the type is not "title", we check whether it is a large description that has one additional line.
+ *
+ * @param {string} text The text that is truncated.
+ * @param {number|null} maximum Optional: a specified maximum length.
```

Comment:
```suggestion
 * @param {number|null} maxCharsOneLine Optional: a specified maximum number of characters on one line.
```

I know this seems to conflict with the constant. You could also set it as a default variable below. I'll write the suggestion.